packages/web3-shared/src/hooks/useResolveEns.ts

```diff
@@ -0,0 +1,10 @@
+import { useAsyncRetry } from 'react-use'
+import { useWeb3 } from './useWeb3'
+
+export function useResolveEns(name: string) {
+    const web3 = useWeb3()
+
+    return useAsyncRetry(async () => {
+        return await web3.eth.ens.getAddress(name)
```

Comment:
+ [ ] Test with Maskbook provider.
+ [ ] Test with MetaMask provider.
+ [ ] Test with WalletConnect provider.

Should a new RPC method be added into `network.ts`?